.github/workflows/functional_test.yml

```diff
@@ -43,16 +43,37 @@ jobs:
         run: go mod download
       - name: Build minikube and e2e test binaries
         run: |
-          make e2e-linux-arm64 e2e-linux-amd64 e2e-darwin-amd64
+          make e2e-linux-amd64
           cp -r test/integration/testdata ./out
       - name: Upload Test Binaries
         uses: actions/upload-artifact@330a01c490aca151604b8cf639adc76d48f6c5d4
         with:
-          name: binaries
+          name: binaries-amd64
+          path: out
+  build-test-binaries-arm64:
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@8e8c483db84b4bee98b60c0593521ed34d9990e8
+      - uses: actions/setup-go@4dc6199c7b1a012772edbd06daecab0f50c9053c
+        with:
+          go-version: ${{env.GO_VERSION}}
+          cache: true      
+      - name: Download Dependencies
+        run: go mod download
+      - name: Build minikube and e2e test binaries
+        run: |
+          make e2e-linux-arm64
+          cp -r test/integration/testdata ./out
+      - name: Upload Test Binaries
+        uses: actions/upload-artifact@330a01c490aca151604b8cf639adc76d48f6c5d4
+        with:
+          name: binaries-arm64
           path: out
   functional-test:
     name: ${{ matrix.name }}
-    needs: build-test-binaries
+    needs:
+      - build-test-binaries-amd64
+      - build-test-binaries-arm64
```

Comment:
If we use matrix, maybe we can depend on single job name? Will make it easier to build more binaries in parallel in the future (darwin-arm64, windows-amd64, windows-arm64).